x-pack/plugins/ml/server/lib/alerts/jobs_health_service.ts

```diff
@@ -345,101 +373,126 @@ export function jobsHealthServiceProvider(
       if (config.mml.enabled) {
         const response = await this.getMmlReport(jobIds);
         if (response && response.length > 0) {
-          const { hard_limit: hardLimitJobs, soft_limit: softLimitJobs } = groupBy(
-            response,
-            'memory_status'
-          );
+          const {
+            hard_limit: hardLimitJobs,
+            soft_limit: softLimitJobs,
+            ok: okJobs,
+          } = groupBy(response, 'memory_status');
 
-          const { count: hardLimitCount, jobsString: hardLimitJobsString } =
-            getJobsAlertingMessageValues(hardLimitJobs);
-          const { count: softLimitCount, jobsString: softLimitJobsString } =
-            getJobsAlertingMessageValues(softLimitJobs);
+          const isHealthy = !hardLimitJobs?.length && !softLimitJobs?.length;
 
           let message = '';
 
-          if (hardLimitCount > 0) {
-            message = i18n.translate('xpack.ml.alertTypes.jobsHealthAlertingRule.mmlMessage', {
-              defaultMessage: `{count, plural, one {Job} other {Jobs}} {jobsString} reached the hard model memory limit. Assign the job more memory and restore from a snapshot from prior to reaching the hard limit.`,
-              values: {
-                count: hardLimitCount,
-                jobsString: hardLimitJobsString,
-              },
-            });
-          }
-
-          if (softLimitCount > 0) {
-            if (message.length > 0) {
-              message += '\n';
-            }
-            message += i18n.translate(
-              'xpack.ml.alertTypes.jobsHealthAlertingRule.mmlSoftLimitMessage',
+          if (isHealthy) {
+            message = i18n.translate(
+              'xpack.ml.alertTypes.jobsHealthAlertingRule.mmlRecoveredMessage',
               {
-                defaultMessage:
-                  '{count, plural, one {Job} other {Jobs}} {jobsString} reached the soft model memory limit. Assign the job more memory or edit the datafeed filter to limit scope of analysis.',
+                defaultMessage: `All jobs are running within configured model memory limits.`,
+              }
+            );
+          } else {
+            const { count: hardLimitCount, jobsString: hardLimitJobsString } =
+              getJobsAlertingMessageValues(hardLimitJobs);
+            const { count: softLimitCount, jobsString: softLimitJobsString } =
+              getJobsAlertingMessageValues(softLimitJobs);
+
+            if (hardLimitCount > 0) {
+              message = i18n.translate('xpack.ml.alertTypes.jobsHealthAlertingRule.mmlMessage', {
+                defaultMessage: `{count, plural, one {Job} other {Jobs}} {jobsString} reached the hard model memory limit. Assign the job more memory and restore from a snapshot from prior to reaching the hard limit.`,
```

Comment:
```suggestion
                defaultMessage: `{count, plural, one {Job} other {Jobs}} {jobsString} reached the hard model memory limit. Assign more memory to the job and restore it from a snapshot taken prior to reaching the hard limit.`,
```

Reply:
FYI this text has been there for a while. It's also been reviewed before